site/src/pages/api/blocks.api.ts

```diff
@@ -105,6 +106,21 @@ export default createApiKeyRequiredHandler<null, ApiSearchResponse>()
         .map(([block, _]) => block);
     }
 
+    for (const block of data) {
+      // Generate absolute URLs for block icons
+      if (block.icon && !block.icon.startsWith("http")) {
+        block.icon = `${FRONTEND_URL}${block.icon}`;
+      }
+
+      for (const variant of block.variants ?? []) {
+        if (variant.icon.startsWith("public")) {
+          variant.icon = `${block.icon!.split("public/")[0]}public/${
+            variant.icon.split("public/")[1]
+          }`;
+        }
+      }
+    }
```

Comment:
I appreciate this is a draft PR but: any relative links in the block package will be relevant to the folder in which the block source code is - e.g. what will now be its `componentId`. You also don't need to treat anything with `public` in it differently - the `public`  folder is in the block's folder, not the root `public` folder.